core/iwasm/interpreter/wasm.h

```diff
@@ -1309,6 +1358,24 @@ wasm_type_equal(const WASMType *type1, const WASMType *type2,
                 const WASMTypePtr *types, uint32 type_count);
 #endif
 
+/* FIXME: move to aot_loader.c if only required by aot */
+inline static bool
+is_valid_func_type(WASMFuncType *func_type)
+{
+    unsigned i;
+    for (i = 0; i < func_type->param_count; i++) {
+        if (!is_valid_value_type(func_type->types[i]))
```

Comment:
Shall we exclude `VALUE_TYPE_VOID`, `PACKED_TYPE_I8` and `PACKED_TYPE_I16`?
param/result type should not be void. And i8/i16 is used only in struct field and array element.

Reply:
in that case, how about using `is_value_type()` in wasm_loader.c, and move it here?

Reply:
Here the function is `inline static bool`, moving is_value_type here may increase footprint, had better move to wasm_runtime_common.h and wasm_runtime_common.c?

Reply:
Move is_value_type to wasm_runtime_common.h/.c and change it to is_valid_value_type, and also add is_valid_func_type in wasm_runtime_common.h/.c?